### PR TITLE
Java bindings

### PR DESCRIPTION
**Pull request by gsmcmullin**
Decode side only.  Working with Android app.

/cc @mookerji

### CODE REVIEW

File: java/README.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
+SBP Bindings for Java
+=======================
+
```
Comment (mookerji) [inline java/README.rst]: Readme here needs an update.

File: java/test/sbp/core_test.clj [hunk 2]
```diff
@@ -0,0 +1,9 @@
+(ns sbp.core-test
```
Comment (mookerji) [inline java/test/sbp/core_test.clj]: Purge the clojure stuff.

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,19 @@
+java/src/com/swiftnav/sbp/acquisition
```
Comment (mookerji) [inline .gitignore]: May want to have a Java-specific .gitignore in the java directory.

File: java/src/com/swiftnav/sbp/drivers/SBPDriverJSSC.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+ */
+package com.swiftnav.sbp.drivers;
+
+import com.swiftnav.sbp.client.SBPDriver;
+
+import java.io.IOException;
+
+import jssc.SerialPort;
+import jssc.SerialPortException;
+
```
Comment (mookerji) [inline java/src/com/swiftnav/sbp/drivers/SBPDriverJSSC.java]: Was this stuff taken from the old JVM PR? If so, we probably ought to credit the forum post by this guy: https://github.com/swift-nav/libsbp/issues/147, where some of the original PR was taken. Unfortunately, the download link from that post doesn't seem to work anymore.

**Comment by mookerji** [comment]
After fixing the merge conflict and removing the detritus, feel free to merge whenever.

**Comment by gsmcmullin** [comment]
retest this please

**Comment by gsmcmullin** [comment]
retest this please

File: java/example/SerialLink.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        try {
+            handler = new SBPHandler(new SBPDriverJSSC(port));
+        } catch (SerialPortException e) {
+            System.err.println("Failed to open serial port: " + e.toString());
+            System.exit(-1);
+        }
+        handler.addCallback(MsgPrintDep.TYPE, printHandler);
+        handler.addCallback(MsgTrackingStateDepA.TYPE, trackingHandler);
+        try {
+            handler.addCallback(new JSONLogger(new FileOutputStream("logfile.json")));
```
Comment (ungureanuvladvictor) [inline java/example/SerialLink.java]: We should handle yyyymmddhhmmss here.

Comment (gsmcmullin) [inline java/example/SerialLink.java]: It probably should, but I thought the point of the example was to demonstrate how to use the library.  Figuring out how to format date strings is off topic.

Comment (ungureanuvladvictor) [inline java/example/SerialLink.java]: if so lgtm

File: java/example/SerialLink.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+
+import jssc.SerialPortException;
+
+public class SerialLink {
+    private SBPHandler handler;
+
+    public static void main(String[] args) {
+        new SerialLink(args[0]);
```
Comment (gsmcmullin) [inline java/example/SerialLink.java]: This crashes if called without args.  Must fix.

**Comment by gsmcmullin** [comment]
retest this please

**Pull request merged**